test/extensions/filters/network/thrift_proxy/decoder_test.cc

```diff
@@ -336,6 +336,28 @@ TEST_P(DecoderStateMachineValueTest, ListValue) {
   EXPECT_EQ(dsm.currentState(), ProtocolState::ListEnd);
 }
 
+TEST_P(DecoderStateMachineValueTest, IncompleteListValue) {
+  FieldType field_type = GetParam();
+  Buffer::OwnedImpl buffer;
+  InSequence dummy;
+
+  EXPECT_CALL(proto_, readListBegin(Ref(buffer), _, _))
+      .WillOnce(DoAll(SetArgReferee<1>(field_type), SetArgReferee<2>(1), Return(true)));
+
+  expectValue(proto_, handler_, field_type, false);
+
+  DecoderStateMachine dsm(proto_, metadata_, handler_);
+
+  dsm.setCurrentState(ProtocolState::ListBegin);
+  EXPECT_EQ(dsm.run(buffer), ProtocolState::WaitForData);
+  EXPECT_EQ(dsm.currentState(), ProtocolState::ListValue);
+
+  expectValue(proto_, handler_, field_type, false);
```

Comment:
I think you add an iteration where the value is present (result = true) to prove that the decoder continues and completes reading the list. (And similar steps in the other tests.)